src/bootstrap/src/utils/tests/mod.rs

```diff
@@ -98,18 +107,46 @@ impl ConfigBuilder {
         self
     }
 
-    pub fn create_config(mut self) -> Config {
-        // Run in dry-check, otherwise the test would be too slow
-        self.args.push("--dry-run".to_string());
+    pub fn with_default_toml_config(mut self, config_toml: &str) -> Self {
```

Comment:
As noted in one of my other comments, let's *always* load `bootstrap.toml` from the test directory, to make sure that we don't ever load the one from `src` (again, an assert for this in `config.rs`) would be nice. If someone uses `with_default_toml_config`, then just overwrite the default `bootstrap.toml` file (which should be empty by default).

Reply:
Hmm, I see, because `get_toml` is not only used for the root config, but also for profiles and includes. Let's:
1) Revert the changes made to `test_include_precedence_over_profile`
2) Instead of asserting that the TOML path comes from tmp, check that it does not come from `src` if we're inside a test. But do not do this in `get_toml` (as there you don't have access to the rest of the config), but do it in `parse_inner`, and check two things:
   - We must have `--config` specified (unless the config might be loaded from the src root)
   - If we do, make sure that the config path is not a child directory of `src`